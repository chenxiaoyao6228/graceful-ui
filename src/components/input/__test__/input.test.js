import { mount } from "@vue/test-utils"
import { render } from "@testing-library/vue";
import Input from "../input.vue";

let InputComponent = {
  components: {Input},
  template: `
        <form>
            <Input data-testid="input" v-model="value" placeholder="请输入内容..."></Input>
            <span data-testid="content">{{value}}</span>
        </form>
    `,
  data() {
    return { value: '1' };
  }
};

describe("Input", () => {
  it("snapshots", () => {
    const { container } = render(InputComponent);
    expect(container).toMatchSnapshot();
  });

  it("v-model", async() => {
    const wrapper = mount(InputComponent)
    const input = wrapper.find('input')
    input.element.value = '123'
    input.trigger('input')
    expect(wrapper.vm.value).toBe('123')
  });
});
