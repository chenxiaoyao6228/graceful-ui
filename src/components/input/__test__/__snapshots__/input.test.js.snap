// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshots 1`] = `
<div>
  <form>
    <div
      class="g-input-wrapper g-input-wrapper"
      data-testid="input"
    >
      <svg
        class="g-icon g-input-icon-clearable"
      >
        <use
          aria-hidden="true"
          xlink:href="#icon-clear"
        />
      </svg>
       
      <input
        class="g-input g-input-default"
        placeholder="请输入内容..."
        type="text"
      />
    </div>
     
    <span
      data-testid="content"
    >
      1
    </span>
  </form>
</div>
`;
